src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BigDecimalPrecisionTest.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.sqlserver.jdbc.unit.statement;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.math.BigDecimal;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import com.microsoft.sqlserver.jdbc.RandomUtil;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+public class BigDecimalPrecisionTest extends AbstractTest {
+
+    String procName1 = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("test_bigdecimal_3"));
+    String procName2 = AbstractSQLGenerator.escapeIdentifier(RandomUtil.getIdentifier("test_bigdecimal_5"));
+
+    @BeforeEach
+    public void init() throws SQLException {
+        try (Connection connection = getConnection()) {
+            String createProcedureSQL1 = "CREATE PROCEDURE " + procName1 + "\n" +
+                    "    @big_decimal_type      decimal(15, 3),\n" +
+                    "    @big_decimal_type_o    decimal(15, 3) OUTPUT\n" +
+                    "AS\n" +
+                    "BEGIN\n" +
+                    "    SET @big_decimal_type_o = @big_decimal_type;\n" +
+                    "END;";
+            String createProcedureSQL2 = "CREATE PROCEDURE " + procName2 + "\n" +
+                    "    @big_decimal_type      decimal(15, 5),\n" +
+                    "    @big_decimal_type_o    decimal(15, 5) OUTPUT\n" +
+                    "AS\n" +
+                    "BEGIN\n" +
+                    "    SET @big_decimal_type_o = @big_decimal_type;\n" +
+                    "END;";
+
+            try (Statement stmt = connection.createStatement()) {
+                stmt.execute(createProcedureSQL1);
+                stmt.execute(createProcedureSQL2);
+            }
+        }
+    }
+
+    @AfterEach
+    public void terminate() throws SQLException {
```

Comment:
Let us use this approach for table creation and dropping in init and terminate methods as it provides idempotency on failures and repeated runs


        @BeforeEach
        public void init() throws Exception {
            try (Connection con = getConnection()) {
                con.setAutoCommit(false);
                try (Statement stmt = con.createStatement()) {
                    TestUtils.dropTableIfExists(tableName, stmt);
                    stmt.executeUpdate(
                            "CREATE TABLE " + tableName + " (col1 INT, col2 VARCHAR(" + TEST_STRING.length() + "))");
                }
                con.commit();
            }
        }



       @AfterEach
        public void terminate() throws Exception {
            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
                try {
                    TestUtils.dropTableIfExists(tableName, stmt);
                } catch (SQLException e) {
                    fail(TestResource.getResource("R_unexpectedException") + e.getMessage());
                }
            }
        }